.github/workflows/ci.yml

```diff
@@ -62,7 +62,57 @@ jobs:
         run: |
           conda list
 
-
       - name: Run Namespace Tests
         run: |
           python -m pytest test
+
+link-check:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v5.0.0
+
+      - name: environment setup
+        id: link-env-setup
+        continue-on-error: true
+        uses: mamba-org/setup-micromamba@0dea6379afdaffa5d528b3d1dabc45da37f443fc # v2
+        with:
+          environment-file: ci/docs.yml
+          cache-environment: true
+          cache-environment-key: "linkcheck-${{env.TODAY}}"
+          create-args: >-
+            python=3.11
+
+      - name: retry environment set up if failed
+        if: steps.link-env-setup.outcome == 'failure'
+        uses: mamba-org/setup-micromamba@0dea6379afdaffa5d528b3d1dabc45da37f443fc # v2
+        with:
+          download-micromamba: false
+          environment-file: ci/docs.yml
+          cache-environment: true
+          cache-environment-key: "linkcheck-${{env.TODAY}}"
+          create-args: >-
+            python=3.11
+
+      - name: Install uxarray
+        run: |
+          python -m pip install . --no-deps
+
+      - name: conda list
+        run: |
+          conda list
+
+      - name: Make docs with linkcheck
+        uses: nick-fields/retry@ce71cc2ab81d554ebbe88c79ab5975992d79ba08 # v3
+        with:
+          timeout_minutes: 60
+          max_attempts: 3
+          command: |
+            eval "$(micromamba shell hook --shell bash)"
+            micromamba activate
```

Comment:
Is this needed? 
You're running micromamba activate twice without specifying an environment the first time. Remove it.
```suggestion
```

Also, the eval .. line might be redundant, as at the top bash shell is already specified.

Reply:
Oh, thanks for catching that! Also regarding `eval` you are probably right; I haven't removed it yet but we can.